server/handler.go

```diff
@@ -115,27 +142,34 @@ func (h *Handler) ComQuery(
 
 			r = nil
 			proccesedAtLeastOneBatch = true
-
 			continue
 		}
 
-		row, err := rows.Next()
-		if err != nil {
+		select {
+		case err := <-errchan:
 			if err == io.EOF {
-				break
+				goto EndFor
```

Comment:
Better is to label the `for` statement and break to that label:

```go
nextRow:
  for {
     …
     select {
        …
        break nextRow
```

Reply:
I like gotos for a little first world anarchy but that's cleaner.